src/main/java/com/fasterxml/jackson/core/io/NumberInput.java

```diff
@@ -152,6 +152,32 @@ public static long parseLong(char[] ch, int off, int len)
         long val = parseInt(ch, off, len1) * L_BILLION;
         return val + (long) parseInt(ch, off+len1, 9);
     }
+    
+    /**
+     * Parses an unsigned long made up of exactly 19 digits.
+     * <p>
+     * It is the callers responsibility to make sure the input is exactly 19 digits.
+     * and fits into a 64bit long by calling {@link #inLongRange(char[], int, int, boolean)}
+     * first.
+     * <p>
+     * Note that input String must NOT contain leading minus sign (even
+     * if {@code negative} is set to true).
+     * 
+     * @param ch Buffer that contains integer value to decode
+     * @param off Offset of the first digit character in buffer
+     * @param negative Whether original number had a minus sign
+     * @return Decoded {@code long} value
+     */
+    public static long parseLong19(char[] ch, int off, boolean negative)
+    {
+        // Note: caller must ensure length is [10, 18]
+        long num = 0L;
+        for (int i = 0; i < 19; i++) {
+            char c = ch[off + i];
+            num = (num * 10) + (c - '0');
+        }
+        return negative ? -num : num;
```

Comment:
Another thing to ensure is that `Long.MIN_VALUE` works (negative range has one more value than positive and this means there's one very special value where negating positive value will not work as expected).

Reply:
This is already covered by `NumberParsingTest#testLongRange`

Reply:
Ok, good, thank you for verifying.